check_brevisone

```diff
@@ -136,54 +133,88 @@ if __name__ == '__main__':
-    
-    args = parser.parse_args()
-    
-    signal.signal(signal.SIGALRM, partial(handle_sigalrm, timeout=args.timeout))
-    signal.alarm(args.timeout)
-    
-    # Execute Check
-    try:
-        if args.insecure is True:
-            requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
 
-        response = requests.get(path % (args.protocol, args.hostname), verify=(not args.insecure))
+    return parser.parse_args(args)
+
 
-        details = response.text.split("\n")
+def get_data(base_url, timeout, insecure):
+    """
+    Requests the data via HTTP
+    Basically a wrapper so that we can mock during tests
 
-        lines = []
-        for item in details:
-            if not item:
-                continue
+    What we expect from the server:
 
-            lines.append(str(item).strip())
+    HTTP/1.0 200 OK
+    que: foo
+    failed: 0
+    signal_strength: 15 db
+    total: 25
+    """
 
-        if response.status_code >= 400:
-            raise IOError("HTTP Request Failed %d." % response.status_code)
-        if not lines[0].strip().startswith('que:'):
-            raise RuntimeError("Unexpected response, invalid check target '%s'?" % args.hostname)
+    # Default context for connection
+    ctx = ssl.create_default_context()
+    if insecure is True:
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
 
-        que = readInt(lines[0])
-        failed = readInt(lines[1])
-        total = readInt(lines[3])
-        signal_strength = readSig(lines[2])
+    url = urljoin(base_url, "check.php")
+    response = urllib.request.urlopen(url=url, timeout=timeout, context=ctx) # pylint: disable=consider-using-with
 
-        perfdata = {
-            'que': que,
-            'failed': failed,
-            'total': total,
-            'signal': signal_strength
-        }
+    if response.getcode() >= 400:
+        raise RuntimeError("Could not get response")
 
-        if failed >= args.fail or signal_strength <= args.signal_critical:
-            output('CRITICAL', 2, lines=lines, perfdata=perfdata)
+    resp = response.read().decode()
+    if "que:" not in resp:
+        raise RuntimeError("Invalid response")
 
-        # Return
-        if que >= args.queue or signal_strength <= args.signal_warning:
-            output('WARNING', 1, lines=lines, perfdata=perfdata)
+    return resp
 
-        output('OK', 0, lines=lines, perfdata=perfdata)
-    except Exception as e:
-        output(e, 2)
+
+def main(args):
+    try:
+        base_url = args.protocol + '://' + args.hostname
+        data = get_data(base_url=base_url,
+                        timeout=args.timeout,
+                        insecure=args.insecure)
+    except Exception as data_exc: # pylint: disable=broad-except
+        print('UNKNOWN - Could not connect to SMS Gateway', data_exc)
+        return UNKNOWN
+
+    # Split up lines and clean up items
+    lines = [str(i).strip() for i in data.split("\n") if i]
```

Comment:
The printing of all lines should remain. I figured that there might be other that's not perfdata

Reply:
Done :heavy_check_mark: